base/io.jl

```diff
@@ -63,11 +61,27 @@ false
 function isopen end
 
 """
-    close(stream)
+    close(io::IO)
+
+Close `io`. Performs a [`flush`](@ref) first.
+
+Closing an IO signals that its underlying resources (OS handle, network
+connections, etc) should be destroyed.
+A closed IO is in an undefined state and should not be written to or read from.
+When attempting to do so, the IO may throw an exception, continue to behave
+normally, or read/write zero bytes, depending on the implementation.
+However, implementations should make sure that reading to or writing from a
+closed IO does not cause undefined behaviour.
 
-Close an I/O stream. Performs a [`flush`](@ref) first.
+This function is generically defined to only `flush` the io. That allows
+wrapping IOs to close its underlying IO.
+
+See also: [`isopen`](@ref)
 """
-function close end
+function close(io::IO)
+    flush(io)
+    nothing
+end
```

Comment:
Isn't this a change in behavior and thus goes beyond what the title of this PR suggests? I feel a bit uncomfortable with this -- I'd immediately agree to merge the docstring changes, but I don't feel qualified to judge whether this change is OK.

Reply:
Agreed. The rest seems good

Reply:
Alright, will remove.